testsuite/tests/basic/patmatch_split_no_or.ml

```diff
@@ -0,0 +1,87 @@
+(* TEST
+ flags = "-nostdlib -nopervasives -dlambda"
+ * expect
+ *)
+
+(******************************************************************************)
+
+(* Check that the extra split indeed happens when the last row is made of
+   "variables" only *)
+
+let last_is_anys = function
+  | true, false -> 1
+  | _, false -> 2
+  | _, _ -> 3
+;;
+[%%expect{|
+(let
+  (last_is_anys/10 =
+     (function param/11 : int
+       (catch
+         (if (field 0 param/11) (if (field 1 param/11) (exit 1) 1)
+           (if (field 1 param/11) (exit 1) 2))
+        with (1) 3)))
+  (apply (field 1 (global Toploop!)) "last_is_anys" last_is_anys/10))
+val last_is_anys : bool * bool -> int = <fun>
+|}]
+
+let last_is_vars = function
+  | true, false -> 1
+  | _, false -> 2
+  | _x, _y -> 3
+;;
+[%%expect{|
+(let
+  (last_is_vars/16 =
+     (function param/19 : int
+       (catch
+         (if (field 0 param/19) (if (field 1 param/19) (exit 3) 1) (exit 3))
+        with (3) (if (field 1 param/19) 3 2))))
+  (apply (field 1 (global Toploop!)) "last_is_vars" last_is_vars/16))
+val last_is_vars : bool * bool -> int = <fun>
+|}]
```

Comment:
I think it would be nice to add a comment to tell that currently the last row is not firing here.